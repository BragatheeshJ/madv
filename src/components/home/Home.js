import React from "react";
import { Link } from "react-router-dom";
import Aboutimg from "../../Images/Aboutus.jpg";
import Contactimg from "../../Images/Contactus.jpg";
import Howtoapplyimg from "../../Images/Howtoapplyimg.jpg";

const Home = () => {
  return (
    <div>
      <div>
        <h1>
          <center className="landingpage">
            <h1 className="h">
              Welcome to <b>M-Advisor</b>
            </h1>
          </center>
        </h1>
        <p className="paragraph text-light">
          Planning to take a Mortgage Loan? Use this Mortgage Loan EMI Calculator
          <br />
          to estimate the EMI amount payable per month to repay the Mortgage
          <br />
          Loan. The Mortgage Loan EMI Calculator is easy to use. All you need to
          <br />
          select is the Loan amount, Loan Tenure. After selecting the Loan
          <br />
          Amount, and The Loan Tenure, the calculator will estimate the monthly
          <br />
          EMI amount and the total Interest payable till the end of the loan
          tenure.
        </p>
      </div>
    </div>
  );
};

export default Home;

