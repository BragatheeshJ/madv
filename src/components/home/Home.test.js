import { render, screen } from "@testing-library/react";
import { renderComponent } from "../../utils/test_helpers";
import Home from "./Home";

describe("Home page", () => {
  test("Should Render Intial Page", () => {
    render(renderComponent(<Home />));
    const text1 = screen.getAllByText(/M-Loan/i);
    expect(text1[0]).toBeInTheDocument();
    const text2 = screen.getAllByText(
      /to estimate the EMI amount payable per month to repay the Mortage/i
    );
    expect(text2[0]).toBeInTheDocument();
  });
  test("should test Intial page buttons", () => {
    render(renderComponent(<Home />));
    expect(
      screen.getByRole("button", { name: "Register" })
    ).toBeInTheDocument();
    expect(screen.getByRole("button", { name: "Login" })).toBeInTheDocument();
  });
  test("Matching the snapshot of the navbar", async () => {
    const { asFragment } = render(renderComponent(<Home />));
    expect(asFragment()).toMatchSnapshot();
  });
});