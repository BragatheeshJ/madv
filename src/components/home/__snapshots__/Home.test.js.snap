// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page Matching the snapshot of the navbar 1`] = `
<DocumentFragment>
  <div>
    <div>
      <h1>
        <center
          class="landingpage"
        >
          <h1
            class="h"
          >
            Welcome to 
            <b>
              M-Advisor
            </b>
          </h1>
        </center>
      </h1>
      <p
        class="paragraph text-light"
      >
        Planning to take a Mortgage Loan? Use this Mortgage Loan EMI Calculator
        <br />
        to estimate the EMI amount payable per month to repay the Mortgage
        <br />
        Loan. The Mortgage Loan EMI Calculator is easy to use. All you need to
        <br />
        select is the Loan amount, Loan Tenure. After selecting the Loan
        <br />
        Amount, and The Loan Tenure, the calculator will estimate the monthly
        <br />
        EMI amount and the total Interest payable till the end of the loan tenure.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
